lib/haml_lint/linter/class_attribute_with_static_value.rb

```diff
@@ -9,11 +9,17 @@ module HamlLint
   # ...over:
   #
   #   %tag{ class: 'class-name' }
+  #
+  # But will allow invalid class names for templating:
+  #
+  #   %tag{ class: '{{ template-var }}' }
   class Linter::ClassAttributeWithStaticValue < Linter
     include LinterRegistry
 
     STATIC_TYPES = [:str, :sym].freeze
 
+    VALID_CLASS_REGEX = /^-?[_a-zA-Z]+[_a-zA-Z0-9-]*$/
```

Comment:
I'm a little worried that this regex won't catch all possible valid class names. That being said though, the worst thing that could happen is that you end up not seeing a lint for some obscure class name. That's probably fine.